src/Xamarin.Android.Build.Tasks/Tasks/CheckForInvalidResourceFileNames.cs

```diff
@@ -17,13 +17,22 @@ public class CheckForInvalidResourceFileNames : Task {
 		public ITaskItem[] Resources { get; set; }
 
 		Regex fileNameCheck = new Regex ("[^a-zA-Z0-9_.]+", RegexOptions.Compiled);
+		Regex fileNameWithHyphenCheck = new Regex ("[^a-zA-Z0-9_.-]+", RegexOptions.Compiled);
```

Comment:
I don't understand the PR; the commit message isn't entirely in sync with the commit contents.

> We forgot to include - as a valid character for the Resource filenames

...so why aren't we just updating `fileNameCheck`? Why is `fileNameWithHypenCheck` required at all?

Reply:
Because after testing, files in the `values-*` folders support `-` but files in other folders (like 'drawable' do not. 

So we have to split the check depending on the folder its in.